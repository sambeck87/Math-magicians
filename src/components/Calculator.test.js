import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import Calc from './Calculator';

test('renders correctly', () => {
  const tree = render(<Calc />);
  expect(tree.toJSON).toMatchSnapshot();
});

describe('testing buttons', () => {
  for (let i = 0; i < 10; i += 1) {
    test(`button ${i}`, () => {
      render(<Calc />);
      fireEvent.click(screen.getByText(`${i}`));
      const testDisplay = screen.getByTestId('result').textContent;
      expect(testDisplay).toMatch(`${i}`);
    });
  }

  test('button +', () => {
    render(<Calc />);
    fireEvent.click(screen.getByText('+'));
    const testDisplay = screen.getByTestId('result').textContent;
    expect(testDisplay).toMatch('+');
  });

  test('button -', () => {
    render(<Calc />);
    fireEvent.click(screen.getByText('-'));
    const testDisplay = screen.getByTestId('result').textContent;
    expect(testDisplay).toMatch('-');
  });

  test('button x', () => {
    render(<Calc />);
    fireEvent.click(screen.getByText('x'));
    const testDisplay = screen.getByTestId('result').textContent;
    expect(testDisplay).toMatch('x');
  });

  test('button รท', () => {
    render(<Calc />);
    fireEvent.click(screen.getByText('รท'));
    const testDisplay = screen.getByTestId('result').textContent;
    expect(testDisplay).toMatch('รท');
  });

  test('button AC', () => {
    render(<Calc />);
    fireEvent.click(screen.getByText('AC'));
    const testDisplay = screen.getByTestId('result').textContent;
    expect(testDisplay).toMatch('');
  });
});
