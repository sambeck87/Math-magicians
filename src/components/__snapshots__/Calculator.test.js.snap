// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator test renders correctly 1`] = `undefined`;

exports[`renders correctly 1`] = `undefined`;
